<a href="https://colab.research.google.com/github/preethisedigar/data_analytics_projects/blob/main/Movies.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In this assignment, you will try to find some interesting insights into a few movies released between 1916 and 2016, using Python. You will have to download a movie dataset, write Python code to explore the data, gain insights into the movies, actors, directors, and collections, and submit the code.

#### Some tips before starting

1. Identify the task to be performed correctly, and only then proceed to write the required code. Don’t perform any incorrect analysis or look for information that isn’t required for the assignment.
2. In some cases, the variable names have already been assigned, and you just need to write code against them. In other cases, the names to be given are mentioned in the instructions. We strongly advise you to use the mentioned names only.
3. Always keep inspecting your data frame after you have performed a particular set of operations.
4. There are some checkpoints given in the IPython notebook provided. They're just useful pieces of information you can use to check if the result you have obtained after performing a particular task is correct or not.
5. Note that you will be asked to refer to documentation for solving some of the questions. That is done on purpose for you to learn new commands and also how to use the documentation.

In [ ]:
# Import the numpy and pandas packages

import numpy as np
import pandas as pd

### Task 1: Reading and Inspection

**Subtask 1.1: Import and read**

Import and read the movie database. Store it in a variable called `movies`.

In [ ]:
# Write your code for importing the csv file here
movies =pd.read_csv("Movies.csv")
movies

,color,director_name,num_critic_for_reviews,duration,director_facebook_likes,actor_3_facebook_likes,actor_2_name,actor_1_facebook_likes,gross,genres,...,num_user_for_reviews,language,country,content_rating,budget,title_year,actor_2_facebook_likes,imdb_score,aspect_ratio,movie_facebook_likes
0,Color,James Cameron,723.0,178.0,0.0,855.0,Joel David Moore,1000.0,760505847.0,Action|Adventure|Fantasy|Sci-Fi,...,3054.0,English,USA,PG-13,237000000.0,2009.0,936.0,7.9,1.78,33000
1,Color,Gore Verbinski,302.0,169.0,563.0,1000.0,Orlando Bloom,40000.0,309404152.0,Action|Adventure|Fantasy,...,1238.0,English,USA,PG-13,300000000.0,2007.0,5000.0,7.1,2.35,0
2,Color,Sam Mendes,602.0,148.0,0.0,161.0,Rory Kinnear,11000.0,200074175.0,Action|Adventure|Thriller,...,994.0,English,UK,PG-13,245000000.0,2015.0,393.0,6.8,2.35,85000
3,Color,Christopher Nolan,813.0,164.0,22000.0,23000.0,Christian Bale,27000.0,448130642.0,Action|Thriller,...,2701.0,English,USA,PG-13,250000000.0,2012.0,23000.0,8.5,2.35,164000
4,Color,Andrew Stanton,462.0,132.0,475.0,530.0,Samantha Morton,640.0,73058679.0,Action|Adventure|Sci-Fi,...,738.0,English,USA,PG-13,263700000.0,2012.0,632.0,6.6,2.35,24000
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
3848,Color,Shane Carruth,143.0,77.0,291.0,8.0,David Sullivan,291.0,424760.0,Drama|Sci-Fi|Thriller,...,371.0,English,USA,PG-13,7000.0,2004.0,45.0,7.0,1.85,19000
3849,Color,Neill Dela Llana,35.0,80.0,0.0,0.0,Edgar Tancangco,0.0,70071.0,Thriller,...,35.0,English,Philippines,Not Rated,7000.0,2005.0,0.0,6.3,NaN,74
3850,Color,Robert Rodriguez,56.0,81.0,0.0,6.0,Peter Marquardt,121.0,2040920.0,Action|Crime|Drama|Romance|Thriller,...,130.0,Spanish,USA,R,7000.0,1992.0,20.0,6.9,1.37,0
3851,Color,Edward Burns,14.0,95.0,0.0,133.0,Caitlin FitzGerald,296.0,4584.0,Comedy|Drama,...,14.0,English,USA,Not Rated,9000.0,2011.0,205.0,6.4,NaN,413


**Subtask 1.2: Inspect the dataframe**

Inspect the dataframe's columns, shapes, variable types etc.

In [ ]:
# Write your code for inspection here
movies.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3853 entries, 0 to 3852
Data columns (total 28 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   color                      3851 non-null   object 
 1   director_name              3853 non-null   object 
 2   num_critic_for_reviews     3852 non-null   float64
 3   duration                   3852 non-null   float64
 4   director_facebook_likes    3853 non-null   float64
 5   actor_3_facebook_likes     3847 non-null   float64
 6   actor_2_name               3852 non-null   object 
 7   actor_1_facebook_likes     3853 non-null   float64
 8   gross                      3853 non-null   float64
 9   genres                     3853 non-null   object 
 10  actor_1_name               3853 non-null   object 
 11  movie_title                3853 non-null   object 
 12  num_voted_users            3853 non-null   int64  
 13  cast_total_facebook_likes  3853 non-null   int64

#### <font color='red'>Question 1: How many rows and columns are present in the dataframe? </font>
-  <font color='red'>(3821, 26)</font>
-  <font color='red'>(3879, 28)</font>
-  <font color='red'>(3853, 28)</font>
-  <font color='red'>(3866, 26)</font>

In [ ]:
movies.shape

(3853, 28)

#### <font color='red'>Question 2: How many columns have null values present in them? Try writing a code for this instead of counting them manually.</font>

-  <font color='red'>3</font>
-  <font color='red'>6</font>
-  <font color='red'>9</font>
-  <font color='red'>12</font>

In [ ]:
a=movies.isnull().any().sum()
a


12

### Task 2: Cleaning the Data

**Subtask 2.1: Drop unecessary columns**

For this assignment, you will mostly be analyzing the movies with respect to the ratings, gross collection, popularity of movies, etc. So many of the columns in this dataframe are not required. So it is advised to drop the following columns.
-  color
-  director_facebook_likes
-  actor_1_facebook_likes
-  actor_2_facebook_likes
-  actor_3_facebook_likes
-  actor_2_name
-  cast_total_facebook_likes
-  actor_3_name
-  duration
-  facenumber_in_poster
-  content_rating
-  country
-  movie_imdb_link
-  aspect_ratio
-  plot_keywords

In [ ]:
# Check the 'drop' function in the Pandas library - dataframe.drop(list_of_unnecessary_columns, axis = )
# Write your code for dropping the columns here. It is advised to keep inspecting the dataframe after each set of operations
movies


,color,director_name,num_critic_for_reviews,duration,director_facebook_likes,actor_3_facebook_likes,actor_2_name,actor_1_facebook_likes,gross,genres,...,num_user_for_reviews,language,country,content_rating,budget,title_year,actor_2_facebook_likes,imdb_score,aspect_ratio,movie_facebook_likes
0,Color,James Cameron,723.0,178.0,0.0,855.0,Joel David Moore,1000.0,760505847.0,Action|Adventure|Fantasy|Sci-Fi,...,3054.0,English,USA,PG-13,237000000.0,2009.0,936.0,7.9,1.78,33000
1,Color,Gore Verbinski,302.0,169.0,563.0,1000.0,Orlando Bloom,40000.0,309404152.0,Action|Adventure|Fantasy,...,1238.0,English,USA,PG-13,300000000.0,2007.0,5000.0,7.1,2.35,0
2,Color,Sam Mendes,602.0,148.0,0.0,161.0,Rory Kinnear,11000.0,200074175.0,Action|Adventure|Thriller,...,994.0,English,UK,PG-13,245000000.0,2015.0,393.0,6.8,2.35,85000
3,Color,Christopher Nolan,813.0,164.0,22000.0,23000.0,Christian Bale,27000.0,448130642.0,Action|Thriller,...,2701.0,English,USA,PG-13,250000000.0,2012.0,23000.0,8.5,2.35,164000
4,Color,Andrew Stanton,462.0,132.0,475.0,530.0,Samantha Morton,640.0,73058679.0,Action|Adventure|Sci-Fi,...,738.0,English,USA,PG-13,263700000.0,2012.0,632.0,6.6,2.35,24000
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
3848,Color,Shane Carruth,143.0,77.0,291.0,8.0,David Sullivan,291.0,424760.0,Drama|Sci-Fi|Thriller,...,371.0,English,USA,PG-13,7000.0,2004.0,45.0,7.0,1.85,19000
3849,Color,Neill Dela Llana,35.0,80.0,0.0,0.0,Edgar Tancangco,0.0,70071.0,Thriller,...,35.0,English,Philippines,Not Rated,7000.0,2005.0,0.0,6.3,NaN,74
3850,Color,Robert Rodriguez,56.0,81.0,0.0,6.0,Peter Marquardt,121.0,2040920.0,Action|Crime|Drama|Romance|Thriller,...,130.0,Spanish,USA,R,7000.0,1992.0,20.0,6.9,1.37,0
3851,Color,Edward Burns,14.0,95.0,0.0,133.0,Caitlin FitzGerald,296.0,4584.0,Comedy|Drama,...,14.0,English,USA,Not Rated,9000.0,2011.0,205.0,6.4,NaN,413


In [ ]:
df= movies.drop(['color','duration','actor_3_facebook_likes','actor_2_name','actor_1_facebook_likes','cast_total_facebook_likes','actor_3_name',
                 'plot_keywords','movie_imdb_link','facenumber_in_poster','content_rating','country','actor_2_facebook_likes','aspect_ratio'], axis=1, errors='ignore')
df.head()

,director_name,num_critic_for_reviews,director_facebook_likes,gross,genres,actor_1_name,movie_title,num_voted_users,num_user_for_reviews,language,budget,title_year,imdb_score,movie_facebook_likes
0,James Cameron,723.0,0.0,760505847.0,Action|Adventure|Fantasy|Sci-Fi,CCH Pounder,Avatar,886204,3054.0,English,237000000.0,2009.0,7.9,33000
1,Gore Verbinski,302.0,563.0,309404152.0,Action|Adventure|Fantasy,Johnny Depp,Pirates of the Caribbean: At World's End,471220,1238.0,English,300000000.0,2007.0,7.1,0
2,Sam Mendes,602.0,0.0,200074175.0,Action|Adventure|Thriller,Christoph Waltz,Spectre,275868,994.0,English,245000000.0,2015.0,6.8,85000
3,Christopher Nolan,813.0,22000.0,448130642.0,Action|Thriller,Tom Hardy,The Dark Knight Rises,1144337,2701.0,English,250000000.0,2012.0,8.5,164000
4,Andrew Stanton,462.0,475.0,73058679.0,Action|Adventure|Sci-Fi,Daryl Sabara,John Carter,212204,738.0,English,263700000.0,2012.0,6.6,24000


#### <font color='red'>Question 3: What is the count of columns in the new dataframe? </font>
-  <font color='red'>10</font>
-  <font color='red'>13</font>
-  <font color='red'>15</font>
-  <font color='red'>17</font>

In [ ]:
df.shape[1]


13

**Subtask 2.2: Inspect Null values**

As you have seen above, there are null values in multiple columns of the dataframe 'movies'. Find out the percentage of null values in each column of the dataframe 'movies'.

In [ ]:
# Write you code here
null_percentage = (df.isnull().sum() / len(df)) * 100
null_percentage

,0
director_name,0.000000
num_critic_for_reviews,0.025954
director_facebook_likes,0.000000
gross,0.000000
genres,0.000000
movie_title,0.000000
num_voted_users,0.000000
num_user_for_reviews,0.000000
language,0.103815
budget,0.000000


#### <font color='red'>Question 4: Which column has the highest percentage of null values? </font>
-  <font color='red'>language</font>
-  <font color='red'>genres</font>
-  <font color='red'>num_critic_for_reviews</font>
-  <font color='red'>imdb_score</font>

In [ ]:
b=null_percentage.idxmax()
print(b)
max(null_percentage)

language


0.10381520892810796

**Subtask 2.3: Fill NaN values**

You might notice that the `language` column has some NaN values. Here, on inspection, you will see that it is safe to replace all the missing values with `'English'`.

In [ ]:
# Write your code for filling the NaN values in the 'language' column here
df["language"] = df["language"].fillna("NAN")
print(df["language"])


0       English
1       English
2       English
3       English
4       English
         ...   
3848    English
3849    English
3850    Spanish
3851    English
3852    English
Name: language, Length: 3853, dtype: object


#### <font color='red'>Question 5: What is the count of movies made in English language after replacing the NaN values with English? </font>
-  <font color='red'>3670</font>
-  <font color='red'>3674</font>
-  <font color='red'>3668</font>
-  <font color='red'>3672</font>

In [ ]:
english_movie_count = (df["language"] == "English").sum()
english_movie_count

3671

### Task 3: Data Analysis

**Subtask 3.1: Change the unit of columns**

Convert the unit of the `budget` and `gross` columns from `$` to `million $`.

In [ ]:
# Write your code for unit conversion her
df['gross'] = df['gross'] / 1_000_000
df['budget'] = df['budget'] / 1_000_000
df

,director_name,num_critic_for_reviews,director_facebook_likes,gross,genres,movie_title,num_voted_users,num_user_for_reviews,language,budget,title_year,imdb_score,movie_facebook_likes
0,James Cameron,723.0,0.0,7.605058e-04,Action|Adventure|Fantasy|Sci-Fi,Avatar,886204,3054.0,English,237.0000,2009.0,7.9,33000
1,Gore Verbinski,302.0,563.0,3.094042e-04,Action|Adventure|Fantasy,Pirates of the Caribbean: At World's End,471220,1238.0,English,300.0000,2007.0,7.1,0
2,Sam Mendes,602.0,0.0,2.000742e-04,Action|Adventure|Thriller,Spectre,275868,994.0,English,245.0000,2015.0,6.8,85000
3,Christopher Nolan,813.0,22000.0,4.481306e-04,Action|Thriller,The Dark Knight Rises,1144337,2701.0,English,250.0000,2012.0,8.5,164000
4,Andrew Stanton,462.0,475.0,7.305868e-05,Action|Adventure|Sci-Fi,John Carter,212204,738.0,English,263.7000,2012.0,6.6,24000
...,...,...,...,...,...,...,...,...,...,...,...,...,...
3848,Shane Carruth,143.0,291.0,4.247600e-07,Drama|Sci-Fi|Thriller,Primer,72639,371.0,English,0.0070,2004.0,7.0,19000
3849,Neill Dela Llana,35.0,0.0,7.007100e-08,Thriller,Cavite,589,35.0,English,0.0070,2005.0,6.3,74
3850,Robert Rodriguez,56.0,0.0,2.040920e-06,Action|Crime|Drama|Romance|Thriller,El Mariachi,52055,130.0,Spanish,0.0070,1992.0,6.9,0
3851,Edward Burns,14.0,0.0,4.584000e-09,Comedy|Drama,Newlyweds,1338,14.0,English,0.0090,2011.0,6.4,413


**Subtask 3.2: Find the movies with highest profit**

   1. Create a new column called `profit` which contains the difference of the two columns: `gross` and `budget`.
   2. Sort the dataframe using the `profit` column as reference. (Find which command can be used here to sort entries from the documentation)
   3. Extract the top ten profiting movies in descending order and store them in a new dataframe - `top10`

In [ ]:
# Write your code for creating the profit column here
df["Profit"] = df["gross"] - df["budget"]
df

,director_name,num_critic_for_reviews,director_facebook_likes,gross,genres,movie_title,num_voted_users,num_user_for_reviews,language,budget,title_year,imdb_score,movie_facebook_likes,Profit
0,James Cameron,723.0,0.0,7.605058e-04,Action|Adventure|Fantasy|Sci-Fi,Avatar,886204,3054.0,English,237.0000,2009.0,7.9,33000,-236.999239
1,Gore Verbinski,302.0,563.0,3.094042e-04,Action|Adventure|Fantasy,Pirates of the Caribbean: At World's End,471220,1238.0,English,300.0000,2007.0,7.1,0,-299.999691
2,Sam Mendes,602.0,0.0,2.000742e-04,Action|Adventure|Thriller,Spectre,275868,994.0,English,245.0000,2015.0,6.8,85000,-244.999800
3,Christopher Nolan,813.0,22000.0,4.481306e-04,Action|Thriller,The Dark Knight Rises,1144337,2701.0,English,250.0000,2012.0,8.5,164000,-249.999552
4,Andrew Stanton,462.0,475.0,7.305868e-05,Action|Adventure|Sci-Fi,John Carter,212204,738.0,English,263.7000,2012.0,6.6,24000,-263.699927
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
3848,Shane Carruth,143.0,291.0,4.247600e-07,Drama|Sci-Fi|Thriller,Primer,72639,371.0,English,0.0070,2004.0,7.0,19000,-0.007000
3849,Neill Dela Llana,35.0,0.0,7.007100e-08,Thriller,Cavite,589,35.0,English,0.0070,2005.0,6.3,74,-0.007000
3850,Robert Rodriguez,56.0,0.0,2.040920e-06,Action|Crime|Drama|Romance|Thriller,El Mariachi,52055,130.0,Spanish,0.0070,1992.0,6.9,0,-0.006998
3851,Edward Burns,14.0,0.0,4.584000e-09,Comedy|Drama,Newlyweds,1338,14.0,English,0.0090,2011.0,6.4,413,-0.009000


In [ ]:
# Write your code for sorting the dataframe here
df_sorted = df.sort_values(by="Profit", ascending=True)
df_sorted

,director_name,num_critic_for_reviews,director_facebook_likes,gross,genres,movie_title,num_voted_users,num_user_for_reviews,language,budget,title_year,imdb_score,movie_facebook_likes,Profit
2678,Joon-ho Bong,363.0,584.0,2.201412e-06,Comedy|Drama|Horror|Sci-Fi,The Host,68883,279.0,Korean,12215.500000,2006.0,7.0,7000,-12215.499998
3282,Chan-wook Park,202.0,0.0,2.116670e-07,Crime|Drama,Lady Vengeance,53508,131.0,Korean,4200.000000,2005.0,7.7,4000,-4200.000000
2693,Lajos Koltai,73.0,45.0,1.958880e-07,Drama|Romance|War,Fateless,5603,45.0,Hungarian,2500.000000,2005.0,7.1,607,-2500.000000
2136,Hayao Miyazaki,174.0,6000.0,2.298191e-06,Adventure|Animation|Fantasy,Princess Mononoke,221552,570.0,Japanese,2400.000000,1997.0,8.4,11000,-2399.999998
2147,Katsuhiro Ôtomo,105.0,78.0,4.103880e-07,Action|Adventure|Animation|Family|Sci-Fi|Thriller,Steamboy,13727,79.0,Japanese,2127.519898,2004.0,6.9,973,-2127.519898
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
3848,Shane Carruth,143.0,291.0,4.247600e-07,Drama|Sci-Fi|Thriller,Primer,72639,371.0,English,0.007000,2004.0,7.0,19000,-0.007000
3850,Robert Rodriguez,56.0,0.0,2.040920e-06,Action|Crime|Drama|Romance|Thriller,El Mariachi,52055,130.0,Spanish,0.007000,1992.0,6.9,0,-0.006998
3845,Olivier Assayas,81.0,107.0,1.360070e-07,Drama|Music|Romance,Clean,3924,39.0,French,0.004500,2004.0,6.9,171,-0.004500
3852,Jon Gunn,43.0,16.0,8.522200e-08,Documentary,My Date with Drew,4285,84.0,English,0.001100,2004.0,6.6,456,-0.001100


In [ ]:
 # Write your code to get the top 10 profiting movies here
print(df_sorted[['movie_title','director_name']].head(10))


                  movie_title    director_name
2678                The Host      Joon-ho Bong
3282          Lady Vengeance    Chan-wook Park
2693                Fateless      Lajos Koltai
2136       Princess Mononoke    Hayao Miyazaki
2147                Steamboy   Katsuhiro Ôtomo
2990                   Akira   Katsuhiro Ôtomo
3629           Godzilla 2000     Takao Okawara
3277                   Tango      Carlos Saura
2743  Kabhi Alvida Naa Kehna       Karan Johar
2890                   Kites       Anurag Basu


**Checkpoint:** You might spot two movies directed by `James Cameron` in the list.

#### <font color='red'>Question 6: Which movie is ranked 5th from the top in the list obtained? </font>
-  <font color='red'>E.T. the Extra-Terrestrial</font>
-  <font color='red'>The Avengers</font>
-  <font color='red'>The Dark Knight</font>
-  <font color='red'>Titanic</font>

In [ ]:
fifth_movie = df_sorted.iloc[-5]
fifth_movie

,3848
director_name,Shane Carruth
num_critic_for_reviews,143.0
director_facebook_likes,291.0
gross,0.0
genres,Drama|Sci-Fi|Thriller
movie_title,Primer
num_voted_users,72639
num_user_for_reviews,371.0
language,English
budget,0.007


**Subtask 3.3: Find IMDb Top 250**

Create a new dataframe `IMDb_Top_250` and store the top 250 movies with the highest IMDb Rating (corresponding to the column: `imdb_score`). Also make sure that for all of these movies, the `num_voted_users` is greater than 25,000.

Also add a `Rank` column containing the values 1 to 250 indicating the ranks of the corresponding films.

In [ ]:
# Write your code for extracting the top 250 movies as per the IMDb score here. Make sure that you store it in a new dataframe
# and name that dataframe as 'IMDb_Top_250'
filtered_df = df[df['num_voted_users'] > 25000]
sorted_df = filtered_df.sort_values(by='imdb_score', ascending=False)
top_250_df = sorted_df.head(250)
top_250_df


,director_name,num_critic_for_reviews,director_facebook_likes,gross,genres,movie_title,num_voted_users,num_user_for_reviews,language,budget,title_year,imdb_score,movie_facebook_likes,Profit
1795,Frank Darabont,199.0,0.0,0.000028,Crime|Drama,The Shawshank Redemption,1689764,4144.0,English,25.0,1994.0,9.3,108000,-24.999972
3016,Francis Ford Coppola,208.0,0.0,0.000135,Crime|Drama,The Godfather,1155770,2238.0,English,6.0,1972.0,9.2,43000,-5.999865
64,Christopher Nolan,645.0,22000.0,0.000533,Action|Crime|Drama|Thriller,The Dark Knight,1676169,4667.0,English,185.0,2008.0,9.0,37000,-184.999467
2543,Francis Ford Coppola,149.0,0.0,0.000057,Crime|Drama,The Godfather: Part II,790926,650.0,English,13.0,1974.0,9.0,14000,-12.999943
325,Peter Jackson,328.0,0.0,0.000377,Action|Adventure|Drama|Fantasy,The Lord of the Rings: The Return of the King,1215718,3189.0,English,94.0,2003.0,8.9,16000,-93.999623
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
67,Jon Favreau,486.0,4000.0,0.000318,Action|Adventure|Sci-Fi,Iron Man,696338,1055.0,English,140.0,2008.0,7.9,10000,-139.999682
73,Doug Liman,585.0,218.0,0.000100,Action|Adventure|Sci-Fi,Edge of Tomorrow,431620,741.0,English,178.0,2014.0,7.9,77000,-177.999900
3052,Yash Chopra,29.0,147.0,0.000003,Drama|Musical|Romance,Veer-Zaara,34449,119.0,Hindi,7.0,2004.0,7.9,2000,-6.999997
95,Peter Jackson,645.0,0.0,0.000303,Adventure|Fantasy,The Hobbit: An Unexpected Journey,637246,1367.0,English,180.0,2012.0,7.9,166000,-179.999697


#### <font color='red'>Question 7: Suppose movies are divided into 5 buckets based on the IMDb ratings: </font>
-  <font color='red'>7.5 to 8</font>
-  <font color='red'>8 to 8.5</font>
-  <font color='red'>8.5 to 9</font>
-  <font color='red'>9 to 9.5</font>
-  <font color='red'>9.5 to 10</font>

<font color = 'red'> Which bucket holds the maximum number of movies from *IMDb_Top_250*? </font>

In [ ]:
top_250_df = pd.DataFrame(top_250_df)
top_250_df
bins = [7.5, 8, 8.5, 9, 9.5, 10]
labels = ['7.5-8', '8-8.5', '8.5-9', '9-9.5', '9.5-10']

In [ ]:
top_250_df['Rating_Bucket'] = pd.cut(top_250_df['imdb_score'], bins=bins, labels=labels, right=False)
bucket_counts = top_250_df['Rating_Bucket'].value_counts()
max_bucket = bucket_counts.idxmax()
max_count = bucket_counts.max()
print(max_bucket )
print(max_count)

8-8.5
159


**Subtask 3.4: Find the critic-favorite and audience-favorite actors**

   1. Create three new dataframes namely, `Meryl_Streep`, `Leo_Caprio`, and `Brad_Pitt` which contain the movies in which the actors: 'Meryl Streep', 'Leonardo DiCaprio', and 'Brad Pitt' are the lead actors. Use only the `actor_1_name` column for extraction. Also, make sure that you use the names 'Meryl Streep', 'Leonardo DiCaprio', and 'Brad Pitt' for the said extraction.
   2. Append the rows of all these dataframes and store them in a new dataframe named `Combined`.
   3. Group the combined dataframe using the `actor_1_name` column.
   4. Find the mean of the `num_critic_for_reviews` and `num_user_for_review` and identify the actors which have the highest mean.

In [ ]:
# Write your code for creating three new dataframes here
Meryl_Streep =  df[df['actor_1_name'] == 'Meryl Streep'] # Include all movies in which Meryl_Streep is the lead
Meryl_Streep.shape[0]

11

In [ ]:
Leo_Caprio = df[df['actor_1_name'] == 'Leonardo DiCaprio']
Leo_Caprio.shape[0]    # Include all movies in which Leo_Caprio is the leadLeo_Caprio

21

In [ ]:
Brad_Pitt = Brad_Pitt = df[df['actor_1_name'] == 'Brad Pitt']
Brad_Pitt.shape[0]  # Include all movies in which Brad_Pitt is the lead

17

In [ ]:
# Write your code for combining the three dataframes here
Combined = pd.concat([Meryl_Streep, Leo_Caprio, Brad_Pitt])
Combined


,director_name,num_critic_for_reviews,director_facebook_likes,gross,genres,actor_1_name,movie_title,num_voted_users,num_user_for_reviews,language,budget,title_year,imdb_score,movie_facebook_likes
392,Nancy Meyers,187.0,278.0,112703470.0,Comedy|Drama|Romance,Meryl Streep,It's Complicated,69860,214.0,English,85000000.0,2009.0,6.6,0
1038,Curtis Hanson,42.0,161.0,46815748.0,Action|Adventure|Crime|Thriller,Meryl Streep,The River Wild,32544,69.0,English,45000000.0,1994.0,6.3,0
1132,Nora Ephron,252.0,0.0,94125426.0,Biography|Drama|Romance,Meryl Streep,Julie & Julia,79264,277.0,English,40000000.0,2009.0,7.0,13000
1322,David Frankel,208.0,64.0,124732962.0,Comedy|Drama|Romance,Meryl Streep,The Devil Wears Prada,286178,631.0,English,35000000.0,2006.0,6.8,0
1390,Robert Redford,227.0,0.0,14998070.0,Drama|Thriller|War,Meryl Streep,Lions for Lambs,41170,298.0,English,35000000.0,2007.0,6.2,0
1471,Sydney Pollack,66.0,521.0,87100000.0,Biography|Drama|Romance,Meryl Streep,Out of Africa,52339,200.0,English,31000000.0,1985.0,7.2,0
1514,David Frankel,234.0,64.0,63536011.0,Comedy|Drama|Romance,Meryl Streep,Hope Springs,34258,178.0,English,30000000.0,2012.0,6.3,0
1563,Carl Franklin,64.0,73.0,23209440.0,Drama,Meryl Streep,One True Thing,9283,112.0,English,30000000.0,1998.0,7.0,592
1784,Stephen Daldry,174.0,335.0,41597830.0,Drama|Romance,Meryl Streep,The Hours,102123,660.0,English,25000000.0,2002.0,7.6,0
2500,Phyllida Lloyd,331.0,58.0,29959436.0,Biography|Drama|History,Meryl Streep,The Iron Lady,82327,350.0,English,13000000.0,2011.0,6.4,18000


In [ ]:
# Write your code for grouping the combined dataframe here
grouped = Combined.groupby('actor_1_name')[['num_critic_for_reviews', 'num_user_for_reviews']].mean()
grouped

,num_critic_for_reviews,num_user_for_reviews
actor_1_name,,
Brad Pitt,245.000000,742.352941
Leonardo DiCaprio,330.190476,914.476190
Meryl Streep,181.454545,297.181818


In [ ]:
# Write the code for finding the mean of critic reviews and audience reviews here
highest_mean_criti = grouped['num_critic_for_reviews'].idxmax()
highest_mean_user = grouped['num_user_for_reviews'].idxmax()



Leonardo DiCaprio
Leonardo DiCaprio


#### <font color='red'>Question 8: Which actor is highest rated among the three actors according to the user reviews? </font>
-  <font color='red'>Meryl Streep</font>
-  <font color='red'>Leonardo DiCaprio</font>
-  <font color='red'>Brad Pitt</font>

In [ ]:
print(highest_mean_criti)

Leonardo DiCaprio


#### <font color='red'>Question 9: Which actor is highest rated among the three actors according to the critics?</font>
-  <font color='red'>Meryl Streep</font>
-  <font color='red'>Leonardo DiCaprio</font>
-  <font color='red'>Brad Pitt</font>

In [ ]:
print(highest_mean_user)

Leonardo DiCaprio
